drivers/rke/rke_driver.go

```diff
@@ -179,13 +182,22 @@ func (d *Driver) Update(ctx context.Context, clusterInfo *types.ClusterInfo, opt
 	defer d.cleanup(stateDir)
 
 	certStr := ""
-	APIURL, caCrt, clientCert, clientKey, certs, err := cmd.ClusterUp(ctx, &rkeConfig, d.DockerDialer, d.LocalDialer,
-		d.wrapTransport(&rkeConfig), false, stateDir, false, false)
+
+	dialers := hosts.GetDialerOptions(d.DockerDialer, d.LocalDialer, d.wrapTransport(&rkeConfig))
+	externalFlags := cluster.GetExternalFlags(false, false, false, false, nil, stateDir, "")
+	if err := cmd.ClusterInit(ctx, &rkeConfig, dialers, externalFlags); err != nil {
+		return nil, err
+	}
+	APIURL, caCrt, clientCert, clientKey, certs, err := cmd.ClusterUp(ctx, dialers, externalFlags)
```

Comment:
shouldn't caCrt, clientcert, clientKey be generated and returned by init?

Reply:
we return these values in clusterUp to make sure that these values are applied to the cluster, Init however generate and populate the desired state of the cluster which still not being applied to the cluster.